firebase-firestore/src/main/java/com/google/firebase/firestore/local/QueryEngine.java

```diff
@@ -61,16 +62,41 @@
 public class QueryEngine {
   private static final String LOG_TAG = "QueryEngine";
 
+  private static final int MIN_COLLECTION_SIZE_TO_AUTO_CREATE_INDEX = 100;
```

Comment:
I took your advise other than `autoIndexCreationRelativeReadCost`, `INDEX_AUTO_CREATION_DEFAULT_RELATIVE_INDEX_READ_COST_PER_DOCUMENT` and `setAutoIndexCreationRelativeIndexReadCost`. Since after changes, the name becomes `indexAutoCreationRelativeIndexReadCostPerDocument`. It is way too long and this cost can potentially used in other place (e.g. decide whether index look up worth it) other than auto creation.